### PR TITLE
Set up base project

### PR DESCRIPTION
**Pull request by junan**
## What happened

Fixed [2](https://github.com/junan/nimble-survey/issues/2)
 
## Insight

- [x] Initialize the project with Angular CLI  `ng new --directory .` command

- [x] Use scss

- [x] Setup testing with Karma, Protractor, Jasmine

 
## Proof Of Work

It will be available in `localhost:4200` after running `ng serve`

Screen Shot 2020-11-16 at 5 24 21 PM

### CODE REVIEW

File: e2e/protractor.conf.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    require('ts-node').register({
+      project: require('path').join(__dirname, './tsconfig.json')
+    });
+    jasmine.getEnv().addReporter(new SpecReporter({
+      spec: {
+        displayStacktrace: StacktraceOption.PRETTY
+      }
+    }));
+  }
+};
```
Comment (carryall) [inline e2e/protractor.conf.js:37]: Can we add an extra line here 🙌

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        { type: 'html' },
+        { type: 'text-summary' }
+      ],
+    },
+    reporters: ['progress', 'kjhtml'],
+    port: 9876,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    autoWatch: true,
+    browsers: ['Chrome'],
```
Comment (carryall) [inline karma.conf.js:31]: How about changing this to `ChromeHeadless` ?

File: e2e/protractor.conf.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * @type { import("protractor").Config }
+ */
+exports.config = {
+  allScriptsTimeout: 11000,
+  specs: [
+    './src/**/*.e2e-spec.ts'
+  ],
+  capabilities: {
+    browserName: 'chrome'
```
Comment (carryall) [inline e2e/protractor.conf.js:16]: How about adding a chrome option here ?
```
chromeOptions: {
      args: ["--headless"]
    }
```

**Review by carryall — CHANGES_REQUESTED** [review]

**Comment by Lahphim** [comment]
@junan Should it be merged to `develop` branch instead of `main` branch?
Screen Shot 2563-11-17 at 10 16 06

File: package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "test": "ng test",
+    "lint": "ng lint",
+    "e2e": "ng e2e"
+  },
+  "private": true,
+  "dependencies": {
+    "@angular/animations": "~11.0.0",
```
Comment (Lahphim) [inline package.json:14]: @junan Should we install all these packages with specific version?

File: src/app/app.component.html [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+<!-- * * * * * * * * * * * The content above * * * * * * * * * * * -->
+<!-- * * * * * * * * * * is only a placeholder * * * * * * * * * * -->
+<!-- * * * * * * * * * * and can be replaced. * * * * * * * * * * * -->
+<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
+<!-- * * * * * * * * * * End of Placeholder * * * * * * * * * * * -->
+<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
+
+
+
+<router-outlet></router-outlet>
```
Comment (Lahphim) [inline src/app/app.component.html:534]: @junan I think we need an tailing line.

File: src/app/app.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    }).compileComponents();
+  });
+
+  it('should create the app', () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    expect(app).toBeTruthy();
+  });
+
+  it(`should have as title 'nimble-survey'`, () => {
```
Comment (Lahphim) [inline src/app/app.component.spec.ts:23]: @junan Should we keep `'` format?

**Review by carryall — APPROVED** [review]

**Comment by junan** [comment]
> @junan Should it be merged to develop branch instead of main branch?
@Lahphim Done

File: src/app/app.component.spec.ts (hunk 6 above)
Comment (junan) [reply src/app/app.component.spec.ts]: @Lahphim Done

File: src/app/app.component.html (hunk 5 above)
Comment (junan) [reply src/app/app.component.html]: @Lahphim Done

File: package.json (hunk 4 above)
Comment (junan) [reply package.json]: @Lahphim Done

**Review by Lahphim — APPROVED** [review]

**Pull request merged**